zne/noise_amplification/folding_amplifier/glorious_local_folding_amplifier.py

```diff
@@ -0,0 +1,170 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022-2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""Glorious Local DAG Folding Noise Amplification (Temporary)"""
+
+
+from collections.abc import Sequence
+
+from qiskit.circuit.library import Barrier, standard_gates
+from qiskit.dagcircuit import DAGCircuit, DAGOpNode
+
+from ..noise_amplifier import DAGNoiseAmplifier
+
+
+class GloriousLocalFoldingAmplifier(DAGNoiseAmplifier):
+    """Amplifies noise in the circuit by said gate and its inverse alternatingly as many times as
+    indicated by noise_factor. The gates that should be folded can be specified by
+    ``gates_to_fold``. By default, all gates of the circuit are folded.
+
+    References:
+        [1] T. Giurgica-Tiron et al. (2020).
+            Digital zero noise extrapolation for quantum error mitigation.
+            `<https://ieeexplore.ieee.org/document/9259940>`
+    """
+
+    def amplify_dag_noise(  # type: ignore # pylint: disable=arguments-differ
+        self,
+        dag: DAGCircuit,
+        noise_factor: float,
+        gates_to_fold: Sequence[int | str] | None,
+    ) -> DAGCircuit:
+        """Applies local folding to input DAGCircuit and returns amplified circuit"""
+        num_full_foldings = self._compute_num_foldings(noise_factor)
+        return self._apply_local_folding(dag, num_full_foldings, gates_to_fold)
+
+        # TODO: Find graceful way of implementing this without overriding typing
+        # TODO: Find way to work with standalone int and str instead of lists
+
+    def _apply_local_folding(
+        self,
+        dag: DAGCircuit,
+        num_foldings: int,
+        gates_to_fold: Sequence[int | str] | None,
+    ) -> DAGCircuit:
+        """Applies local folding strategy and returns a noise amplified dag.
+
+        Args:
+            dag (DAGCircuit): The original dag circuit without foldings.
+            num_foldings (float): Number of times the circuit should be folded.
+            gates_to_fold (Sequence[int | str] | None) : Original gates_to_fold input.
+
+        Returns:
+            DAGCircuit: The noise amplified DAG circuit.
+        """
+        self._validate_gates_to_fold(gates_to_fold)
+        if gates_to_fold in [0, "0", None]:
+            return dag
```

Comment:
This is why you would like to normalize `gates_to_fold`. Any of those inputs should be normalized to `[]`, and this check would not be necessary to begin with 😉  

Nonetheless, for performance reasons it may be convenient to keep:
```suggestion
        if not gates_to_fold:
            return dag
```
which is easier to write/read and also faster to run! 🏃🏼‍♂️